examples/auth.rs

```diff
@@ -0,0 +1,294 @@
+use std::fmt;
+use std::os::raw::c_int;
+use valkey_module::alloc::ValkeyAlloc;
+use valkey_module::{
+    valkey_module, Context, Status, ValkeyError, ValkeyString, AUTH_HANDLED, AUTH_NOT_HANDLED,
+};
+
+#[derive(Debug)]
+enum AuthResult {
+    Allow,
+    Deny,
+    Next,
+}
+
+#[derive(Debug)]
+struct AuthPrivData {
+    result: AuthResult,
+}
+
+// Implement Display for AuthResult
+impl fmt::Display for AuthResult {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AuthResult::Allow => write!(f, "Allow"),
+            AuthResult::Deny => write!(f, "Deny"),
+            AuthResult::Next => write!(f, "Next"),
+        }
+    }
+}
+
+fn authenticate_user(
+    ctx: &Context,
+    username: &ValkeyString,
+    password: &ValkeyString,
+    expected_user: &str,
+    expected_pass: &str,
+    auth_type: &str,
+) -> Result<c_int, ValkeyError> {
+    ctx.log_debug(&format!(
+        "{} auth attempt for user: {}",
+        auth_type,
+        username.to_string()
+    ));
+
+    if username.to_string() == expected_user && password.to_string() == expected_pass {
+        ctx.log_debug(&format!("Matched {} user credentials", auth_type));
+        match ctx.authenticate_client_with_acl_user(username) {
+            Status::Ok => {
+                ctx.log_debug(&format!("Successfully authenticated {} user", auth_type));
```

Comment:
can you make all logging statement `ctx.log_notice`?  This way when running Valkey with default logging settings they show up